src/hooks/useApprove.ts

```diff
@@ -27,22 +31,94 @@ export const useApprove = (lpContract: Contract) => {
 }
 
 // Approve a Pool
-export const useSousApprove = (lpContract: Contract, sousId) => {
+export const useSousApprove = (lpContract: Contract, sousId, earningTokenSymbol) => {
+  const [requestedApproval, setRequestedApproval] = useState(false)
+  const { toastSuccess, toastError } = useToast()
+  const { t } = useTranslation()
   const dispatch = useAppDispatch()
   const { account } = useWeb3React()
   const sousChefContract = useSousChef(sousId)
 
   const handleApprove = useCallback(async () => {
     try {
+      setRequestedApproval(true)
       const tx = await approve(lpContract, sousChefContract, account)
       dispatch(updateUserAllowance(sousId, account))
-      return tx
+      if (tx) {
+        toastSuccess(
+          t('Contract Enabled'),
+          t('You can now stake in the %symbol% pool!', { symbol: earningTokenSymbol }),
+        )
+        setRequestedApproval(false)
+      } else {
+        // user rejected tx or didn't go thru
+        toastError(
+          `${t('Error')}`,
+          `${t(`Please try again. Confirm the transaction and make sure you are paying enough gas!`)}`,
+        )
+        setRequestedApproval(false)
+      }
     } catch (e) {
-      return false
+      console.error(e)
+      toastError('Error', e?.message)
     }
-  }, [account, dispatch, lpContract, sousChefContract, sousId])
+  }, [account, dispatch, lpContract, sousChefContract, sousId, earningTokenSymbol, t, toastError, toastSuccess])
 
-  return { onApprove: handleApprove }
+  return { handleApprove, requestedApproval }
+}
+
+// Approve CAKE auto pool
+export const useVaultApprove = (setLastUpdated: () => void) => {
+  const { account } = useWeb3React()
+  const [requestedApproval, setRequestedApproval] = useState(false)
+  const { t } = useTranslation()
+  const { toastSuccess, toastError } = useToast()
+  const cakeVaultContract = useCakeVaultContract()
+  const cakeContract = useCake()
+
+  const handleApprove = () => {
+    cakeContract.methods
+      .approve(cakeVaultContract.options.address, ethers.constants.MaxUint256)
+      .send({ from: account })
+      .on('sending', () => {
+        setRequestedApproval(true)
+      })
+      .on('receipt', () => {
+        toastSuccess(t('Contract Enabled'), t('You can now stake in the %symbol% vault!', { symbol: 'CAKE' }))
+        setLastUpdated()
```

Comment:
(following our slack convos)

This `setLastUpdated` call is used to manually trigger a state change that used to cause the `checkApprovalStatus`  function to re-fire (not particularly elegant, I know).

On ln 106, you're instantiating a new `lastUpdated` & `setLastUpdated`, which I don't think will be responsive to `setLastUpdated()` being called here, on ln 88.

The impact for a user is that they have to refresh to trigger the vault approval check. Until they do that - they'll be able to endlessly submit approval transactions.

Reply:
I think I've tested approving/disapproving multiple times after this change and it worked fine. Will double-check it once nodes become responsive.....

Reply:
Seems to be fine. I think all those other hooks involved in the process somehow "cache" the state 😄 
https://user-images.githubusercontent.com/81824236/119792445-5e5b7000-bede-11eb-8e7d-ec41c67bf620.mp4

Reply:
👍 nice one, ah yea it could be because these are being called within hooks. yolo